File: infra/cdk/cdk/s3_stack.py
from aws_cdk import core as cdk
from aws_cdk import aws_s3 as s3


class S3Stack(cdk.Stack):

    def __init__(self, scope: cdk.Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        # The code that defines your stack goes here
        # frontend bucket
        artifacts_bucket = s3.Bucket(self, 'frontend',
                                     access_control=s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
                                     encryption=s3.BucketEncryption.S3_MANAGED,
                                     public_read_access=True,
                                     bucket_name='soydecai.xyz',
                                     removal_policy=cdk.RemovalPolicy.DESTROY,
                                     website_index_document='index.html'
                                     )

        cdk.CfnOutput(self, 'frontend-bucket-export',
                      value=artifacts_bucket.bucket_name,
                      export_name='frontend-bucket')

File: ts-infra/lib/s3-stack.ts
import * as cdk from '@aws-cdk/core';
import {Bucket} from '@aws-cdk/aws-s3'
import * as s3 from "@aws-cdk/aws-s3"
import {CfnOutput} from "@aws-cdk/core"


export class S3Stack extends cdk.Stack {
    bucket: Bucket;
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here
    const bucket = new Bucket(this, 'frontendBucket', {
      bucketName:"messages.soydecai.xyz",
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
      websiteIndexDocument:'index.html',
      publicReadAccess: true
    })

      this.bucket = bucket;

    new CfnOutput(
        this,
        'frontEndBucket',
        {
          value:bucket.bucketName,
          exportName: "frontendBucket"
        }
    )
      new CfnOutput(
        this,
        'frontEndBucketUrl',
        {
          value: bucket.bucketWebsiteUrl,
          exportName: "frontendBucketUrl"
        }
    )
      new CfnOutput(
        this,
        'converstaionsUrl',
        {
          value: bucket.bucketWebsiteUrl + "/chats.html",
          exportName: "converstationsUrl"
        }
    )

  }
}
